src/hooks/useRecentSearches.ts

```diff
@@ -1,41 +1,49 @@
 import { useCallback, useEffect, useRef, useState } from 'react';
 import RecentSearches, { ISearch } from 'recent-searches';
 
-export const RECENT_SEARCHES_KEY = '__yxt_recent_searches__';
-
 export function useRecentSearches(
-  recentSearchesLimit: number
-): [ISearch[]|undefined, (input: string) => void, () => void] {
+  recentSearchesLimit: number,
+  verticalKey: string | null
+): [ISearch[] | undefined, (input: string) => void, () => void] {
   const recentSearchesLimitRef = useRef(recentSearchesLimit);
-  const [ recentSearches, setRecentSeaches ] = useState<RecentSearches>(
+  const [recentSearchesKey, setRecentSearchesKey] = useState(getRecentSearchesKey(verticalKey));
```

Comment:
can you look into removing the `useState` for recentSearchesKey and also the `useRef` for recentSearchesLimit? I think since the effect already updates the RecentSearches instance in state we probably don't need them?

Reply:
removed! they work fine without them!